components/server/ee/src/prebuilds/bitbucket-app.ts

```diff
@@ -132,12 +152,23 @@ export class BitbucketApp {
                     data.commitHash,
                 );
             }
-            // todo@alex: add branch and project args
             const ws = await this.prebuildManager.startPrebuild(
                 { span },
                 { user, project: projectAndOwner.project, context, commitInfo },
             );
+            await this.webhookEvents.updateEvent(event.id, {
+                prebuildStatus: "prebuild_triggered",
+                status: "processed",
+                prebuildId: ws.prebuildId,
+            });
             return ws;
+        } catch (e) {
+            console.error("Error processing Bitbucket webhook event", e);
+            await this.webhookEvents.updateEvent(event.id, {
+                prebuildStatus: "prebuild_trigger_failed",
```

Comment:
Note: atm this also gathers a lot of other (potential) errors, but this is a good first intermittent step to untangle this code. :lotus_position: